server.js

```diff
@@ -0,0 +1,68 @@
+const express = require('express');
+
+const app = express();
+const puppeteer = require('puppeteer');
+
+const twitterUrl = 'https://twitter.com/';
+
+const mergeArrayIntoSet = (arr, set) => {
+  arr.forEach((e) => set.add(e))
+}
+
+const getTweets = async (page) => {
+  const data = await page.evaluate(() => {
+    const timeline = document.querySelector("section div");
+    const content = timeline.querySelectorAll("article div[lang][dir]");
+    const lastElement = content[content.length - 1];
+    const tweetsText = Array.from(content, e => e.textContent.trim());
+    lastElement.scrollIntoView();
```

Comment:
I'm curious how difficult it would be to fetch the first n pages of the timeline. If not too hard, could you hardcode this function to scrape 10 pages in one go?

Reply:
The issue is that the tweets fetched in one scroll is variable. So sometimes you scroll, scrape, and get 2 tweets and sometimes you get 7 tweets. 

![scroll](https://user-images.githubusercontent.com/62493636/147875223-121907d2-bbef-4649-931b-3d9244ae18c8.png) 

The number of scrolls is hardcoded (I have set it to 15 now) 

Another alternative is to try to scrape a specific number of tweets. So the scraper will scrape till we have 50 tweets. This can also be added as an input field on the front-end, so the user can specify the number of tweets they want

Reply:
interesting. yes hardcoding to 15 is a good fix for what i wanted to try. Thank you!